coral-service/src/main/java/com/linkedin/coral/coralservice/utils/TranslationUtils.java

```diff
@@ -29,8 +31,21 @@ public static String translateHiveToTrino(String query) {
   }
 
   public static String translateHiveToSpark(String query) {
-    RelNode relNode = new HiveToRelConverter(hiveMetastoreClient).convertSql(query);
-    CoralSpark coralSpark = CoralSpark.create(relNode);
-    return coralSpark.getSparkSql();
+    HiveToRelConverter hiveToRelConverter = new HiveToRelConverter(hiveMetastoreClient);
```

Comment:
(Pl read the comment on CoralSpark.getCoralSqlNode(..) before this comment) 

I think this class could use a bit of refactoring.
For `SqlNode sqlNode = hiveToRelConverter.toSqlNode(query)` : hiveToRelConverter could return a coralSqlNode. Currently, on the LHS side, hiveSqlNode is equivalent CoralSqlNode (LHS).

Inside the if condition, you could extract CoralSelectSqlNode from coralSqlNode, and hand it over first to  `hiveToRelConverter` to generate CoralRelNode. And then hand over sqlCommand + CoralRelNode to CoralSpark.